src/containers/ListingPage/components/ToggleFilterChoices.jsx

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2017-present, NDLA.
+ *
+ *  This source code is licensed under the GPLv3 license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import {
+    addEventListenerForResize,
+    updateIFrameDimensions,
+    addAsideClickListener,
+    removeEventListenerForResize,
+    removeAsideClickListener,
+} from 'ndla-article-scripts';
+
+import React, { PropTypes, Component } from 'react';
+import { ListingShape } from '../../../shapes';
+import FilterChoices from './FilterChoices';
+
+
+class ToggleFilterChoices extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isVisFilter: false,
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  componentDidMount() {
+    addEventListenerForResize();
+    updateIFrameDimensions();
+    addAsideClickListener();
+  }
+
+  componentWillUnmount() {
+    removeEventListenerForResize();
+    removeAsideClickListener();
+  }
+
+  handleClick() {
+    this.setState({ isVisFilter: !this.state.isVisFilter });
```

Comment:
Når du skal basere deg på forrige state burde det gjøres slik: `this.setState(prevState => ({ isVisFilter: !prevState.isVisFilter }));`. Dette for å forhindre feil ifh. til asynkronitet.